### PR TITLE
Case/coverage

### PR DESCRIPTION
**Pull request by titor-oopart**

### CODE REVIEW

File: src/main/java/com/foundation/salesforce/entities/Context.java [hunk 1]
```diff
@@ -26,6 +26,7 @@
     private Lead lead;
     private Response response;
     private Account account;
+    private Case fCase;
```
Comment (MelissaRoman06) [inline src/main/java/com/foundation/salesforce/entities/Context.java]: please give a more descriptive name to the variable.

File: src/main/java/com/foundation/salesforce/core/api/CaseApi.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+ * disclose such Confidential Information and shall use it only in
+ * accordance with the terms of the license agreement you entered into
+ * with Jala Foundation.
+ */
+
+package com.foundation.salesforce.core.api;
+
+import com.foundation.salesforce.core.restClient.Authentication;
+import com.foundation.salesforce.core.restClient.RestClientApi;
+import io.restassured.response.Response;
```
Comment (TheYisusGithub) [inline src/main/java/com/foundation/salesforce/core/api/CaseApi.java]: Please put space between imports.

File: src/test/java/com/foundation/salesforce/steps/CaseStep.java [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+ * with Jala Foundation.
+ */
+
+package com.foundation.salesforce.steps;
+
+import com.foundation.salesforce.core.api.CaseApi;
+import com.foundation.salesforce.core.restClient.RestClientApi;
+import com.foundation.salesforce.core.utils.ResponseValidation;
+import com.foundation.salesforce.entities.Context;
+import cucumber.api.java.en.And;
```
Comment (TheYisusGithub) [inline src/test/java/com/foundation/salesforce/steps/CaseStep.java]: Please add one space between imports.

File: src/test/java/com/foundation/salesforce/steps/CaseStep.java [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+import cucumber.api.java.en.Given;
+import cucumber.api.java.en.Then;
+import cucumber.api.java.en.When;
+import io.restassured.response.Response;
+import io.restassured.response.ValidatableResponse;
+import io.restassured.specification.RequestSpecification;
+import org.json.JSONObject;
```
Comment (TheYisusGithub) [inline src/test/java/com/foundation/salesforce/steps/CaseStep.java:29]: Please add space between imports.

File: src/test/java/com/foundation/salesforce/steps/CaseStep.java [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+     */
+    @Then("the status code is a numbert {int}")
+    public void theStatusCodeIs(int statusCode) {
+        Assert.assertEquals(response.getStatusCode(), statusCode);
+    }
+
+    /**
+     * Schema case validation.
+     */
+    @And("schema case {string} is valid")
```
Comment (MelissaRoman06) [inline src/test/java/com/foundation/salesforce/steps/CaseStep.java]: There is already a method for schema validation in common steps.

File: src/test/resources/schemas/CaseCreate201Schema.json [hunk 6]
```diff
@@ -0,0 +1,7 @@
+{
+  "id": "string",
+  "errors": [
+    "string"
+  ],
+  "success": "boolean"
+}
```
Comment (TheYisusGithub) [inline src/test/resources/schemas/CaseCreate201Schema.json]: Please add salt of line in this place.

File: src/main/java/com/foundation/salesforce/entities/Context.java (hunk 1 above)
Comment (titor-oopart) [reply src/main/java/com/foundation/salesforce/entities/Context.java]: done

File: src/test/java/com/foundation/salesforce/steps/CaseStep.java (hunk 5 above)
Comment (titor-oopart) [reply src/test/java/com/foundation/salesforce/steps/CaseStep.java]: done

File: src/main/java/com/foundation/salesforce/core/api/CaseApi.java (hunk 2 above)
Comment (titor-oopart) [reply src/main/java/com/foundation/salesforce/core/api/CaseApi.java]: done

File: src/test/java/com/foundation/salesforce/steps/CaseStep.java (hunk 3 above)
Comment (titor-oopart) [reply src/test/java/com/foundation/salesforce/steps/CaseStep.java]: done

File: src/test/resources/schemas/CaseCreate201Schema.json (hunk 6 above)
Comment (titor-oopart) [reply src/test/resources/schemas/CaseCreate201Schema.json]: done

File: src/test/java/com/foundation/salesforce/steps/CaseStep.java (hunk 4 above)
Comment (titor-oopart) [reply src/test/java/com/foundation/salesforce/steps/CaseStep.java:29]: done

File: src/main/java/com/foundation/salesforce/core/api/CaseApi.java [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+     */
+    private String finalEndpoint;
+
+    /**
+     * Variable for the response.
+     */
+    private Response response;
+
+    /**
+     * Constructor of AccountAPI.
```
Comment (Cxristian) [inline src/main/java/com/foundation/salesforce/core/api/CaseApi.java]: This is not account api modify all comments

File: src/test/resources/features/case/CreateCase.feature [hunk 8]
```diff
@@ -0,0 +1,112 @@
+Feature: Creation funtional and acceptance
```
Comment (Cxristian) [inline src/test/resources/features/case/CreateCase.feature:2]: tags missing

File: src/test/resources/features/case/CreateCase.feature [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+      | Reason               | Origin    | Status    | Priority |
+      | Installation         | Phone     | New       | Medium   |
+      | Equipment Complexity | Email     | New       | Medium   |
+      | Performance          | Web       | Escalated | High     |
+      | Breakdown            | Web       | Working   | Low      |
+      | Equipment Design     | Web       | Escalated | Low      |
+      | Feedback             | --None--  | --None--  | --None-- |
+      | Other                | Phone     | Escalated | High     |
+      | --None--             | Phone     | Escalated | High     |
+
```
Comment (Cxristian) [inline src/test/resources/features/case/CreateCase.feature]: add line blank

File: src/main/java/com/foundation/salesforce/core/api/CaseApi.java [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+     */
+    protected CaseApi() {
+        restClient = RestClientApi.getInstance();
+        restClient.setRequest(Authentication.requestSpecification());
+    }
+
+    /**
+     * Returns the instance from account API.
```
Comment (alszla) [inline src/main/java/com/foundation/salesforce/core/api/CaseApi.java]: Please replace account API for case API in this documentation.

File: src/main/java/com/foundation/salesforce/core/api/CaseApi.java [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+     *
+     * @return a account API.
```
Comment (alszla) [inline src/main/java/com/foundation/salesforce/core/api/CaseApi.java]: it should say 
@return a case API.

File: src/main/java/com/foundation/salesforce/core/api/CaseApi.java (hunk 7 above)
Comment (titor-oopart) [reply src/main/java/com/foundation/salesforce/core/api/CaseApi.java]: done

File: src/test/resources/features/case/CreateCase.feature (hunk 8 above)
Comment (titor-oopart) [reply src/test/resources/features/case/CreateCase.feature:2]: done

File: src/test/resources/features/case/CreateCase.feature (hunk 9 above)
Comment (titor-oopart) [reply src/test/resources/features/case/CreateCase.feature]: done

File: src/main/java/com/foundation/salesforce/core/api/CaseApi.java (hunk 10 above)
Comment (titor-oopart) [reply src/main/java/com/foundation/salesforce/core/api/CaseApi.java]: done

File: src/main/java/com/foundation/salesforce/core/api/CaseApi.java (hunk 11 above)
Comment (titor-oopart) [reply src/main/java/com/foundation/salesforce/core/api/CaseApi.java]: done

**Review by alszla — APPROVED** [review]

**Review by MelissaRoman06 — APPROVED** [review]

**Pull request merged**